tests/apps/pubsub-publisher/app.go

```diff
@@ -464,7 +464,7 @@ func performPublishHTTP(reqID string, topic string, jsonValue []byte, contentTyp
 	if len(metadata) > 0 {
 		params := netUrl.Values{}
 		for k, v := range metadata {
-			params.Set(fmt.Sprintf("metadata.%s", k), v)
+			params.Set("metadata.%s"+k, v)
```

Comment:
```suggestion
			params.Set("metadata."+k, v)
```
@cicoyle Without looking too much into it, this might be the cause for the e2e failures

Reply:
Yup, just pushed a fix :)